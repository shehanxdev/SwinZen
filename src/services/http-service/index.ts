export * from './APIError';
export * from './HttpService';
export * from './HttpServiceInstance';
