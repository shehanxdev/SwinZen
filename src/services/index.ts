export * from './config-service';
export * from './http-service';
export * from './jwt-decode-service';
export * from './linking-service';
export * from './navigation-service';
export * from './rest-service';
export * from './toast-service';
