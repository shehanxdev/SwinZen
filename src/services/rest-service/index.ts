export * from './account';
export * from './auth';
export * from './info';
export * from './pricePlan';
