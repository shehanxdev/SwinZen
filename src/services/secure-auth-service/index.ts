export * from './secure-auth.service';
export * from './secure-auth-error';
