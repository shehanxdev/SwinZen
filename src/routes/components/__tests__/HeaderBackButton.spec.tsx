import { fireEvent, render } from '@testing-library/react-native';
import * as React from 'react';
import { act } from 'react-test-renderer';

import { NavigationService } from '@sz/services';

import { HeaderBackButton } from '../HeaderBackButton';

describe('HeaderBackButton', () => {
  const TEST_ID = 'HeaderBackButton';

  const getRenderedComponent = () => render(<HeaderBackButton />);

  it(`should render correctly`, () => {
    expect(getRenderedComponent().toJSON()).toMatchSnapshot();
  });

  it('should call NavigationService.goBack on press', async () => {
    const { getByTestId } = getRenderedComponent();
    const button = getByTestId(TEST_ID);

    const mockGoBack = jest.spyOn(NavigationService, 'goBack');

    await act(() => {
      fireEvent.press(button);
    });

    expect(mockGoBack).toHaveBeenCalledTimes(1);
  });
});
