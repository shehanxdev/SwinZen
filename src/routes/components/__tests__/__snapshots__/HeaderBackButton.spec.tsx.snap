// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "paddingBottom": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
  testID="HeaderBackButton"
>
  <SvgMock
    height={20}
    width={12}
  />
</View>
`;
